### PR TITLE
Because products with zero pre-sale prices, are probably not considered to be on sale

### PR DESCRIPTION
**Pull request by craigjbass**

### CODE REVIEW

File: app/models/concerns/pre_sale_prices/product_pre_saleable.rb [hunk 1]
```diff
@@ -13,8 +13,8 @@ module ProductPreSaleable
       delegate :pre_sale_price, to: :master
     end
 
-    def on_sale?
-      pre_sale_prices.count > 0
+    def on_sale?(currency)
+      pre_sale_prices.in_currency(currency).sum(:amount) > 0
```
Comment (nickwood) [inline app/models/concerns/pre_sale_prices/product_pre_saleable.rb:17]: what happens if there are no pre_sale_prices?

File: spec/models/product_spec.rb [hunk 2]
```diff
@@ -1,23 +1,55 @@
...
+  end
+
+  context 'when the pre-sale price is zero in GBP' do
+    let(:zero_gbp_price) { create(:pre_sale_price, amount: 0, currency: 'GBP') }
+    let(:variant_with_zero_pre_sale_price) { create(:variant, pre_sale_prices: [zero_gbp_price]) }
+    let(:product_with_zero_presale_price) { create(:product, master: variant_with_zero_pre_sale_price) }
+
+    subject { product_with_zero_presale_price.on_sale?('GBP') }
+
+    it { is_expected.to be(false) }
```
Comment (chrisblackburn) [inline spec/models/product_spec.rb]: We're using `be_truthy` and `be_falsy` elsewhere in this spec. Any reason for the change?

**Comment by chrisblackburn** [comment]
👍

File: app/models/concerns/pre_sale_prices/product_pre_saleable.rb (hunk 1 above)
Comment (craigjbass) [reply app/models/concerns/pre_sale_prices/product_pre_saleable.rb:17]: @nickwood https://github.com/madetech/spree_pre_sale_prices/pull/6/commits/13bc5d098b3dc6f1a9b42e98334aa582362819f0 this happens

File: spec/models/product_spec.rb (hunk 2 above)
Comment (craigjbass) [reply spec/models/product_spec.rb]: No reason whatsoever

**Comment by emileswarts** [comment]
👍

**Pull request merged**